### ISSUE TITLE
additing reactions to RMG database

### ISSUE THREAD

**Issue by ysadat7** [ReactionMechanismGenerator/RMG-Py#2084]
### Topic
General area which your question is related to.
 - [ ] Installation of RMG
 - [ ] Running an RMG job
 - [ ] Using RMG API
 - [ ] Arkane (formerly CanTherm)
 - [ ] Dependencies
 - [ ] An error message 

### Context
A clear and concise description what you're trying to do.

### Question
Write your question here.

### Installation Information
Describe your installation method and system information if applicable.
 - OS (include version if known): (e.g. Windows 10, Ubuntu 18, macOS 10.13, etc.)
 - Installation method: (e.g. installation from binary or source, with or without anaconda)
 - RMG version information:
   
   - RMG-Py: 
   - RMG-database:

**Comment by ysadat7** [comment]
I am using RMG to generate a chemical kinetic mechanism for the autoxidation reactions in jet fuel. That means, I am. running it in the liquid phase. With the limitations of the liquid phase, RMG does not include a very important reaction which the decomposition of Hydroperoxide: ROOH -> RO + OH . I need to add this reaction and possibly some other related reactions to the back end of RMG.

I was wondering if this is at all possible? And if so, how practical it would be to force RMG to recognise that reaction at the early stages of the mechanism generation.

Many thanks

**Comment by mjohnson541** [comment]
That reaction should be generated by RMG through the R_Recombination family if it's included in the database block (it should be included if you're using "default"). You can force specific species to be included in your run by including them in the input file. However, not knowing what's in your input file or log file, my best guess is the most likely reason you're missing it is because your toleranceMoveToCore isn't small enough.

**Comment by ysadat7** [comment]
Thanks Matt. This is an example of my input file and I have tried to include ROOH, RO radical and OH radical in the run but I'm still not getting the decomposition of ROOH in the output file. Please feel free to have a look at my input file. One more thing, when you say include the species in the input file, what concentration shall I put them in, do you think this is reasonable? 1.00000000E-10,'mol/cm^3'

Input file:

# Data sources
database(
    thermoLibraries = ['primaryThermoLibrary'],
    reactionLibraries = [],
    seedMechanisms = [],
    kineticsDepositories = ['training'],
    kineticsFamilies = 'default',
    kineticsEstimator = 'rate rules',
)

# List of species                            fuel, ao, and smiles

species(
    label='decane',
    reactive=True,
    structure=SMILES("CCCCCCCCCC"),
)
species(
    label='oxygen',
    reactive=True,
    structure=SMILES("[O][O]"),
)
species(
    label='ROOH',
    reactive=True,
    structure=SMILES("CCCCCCCCCCOO"),
)
species(
    label='RO',
    reactive=True,
    structure=SMILES("CCCCCCCCCC[O]"),
)
species(
    label='OH',
    reactive=True,
    structure=SMILES("[OH]"),
)
# Reaction systems
liquidReactor(
    temperature=(423,'K'),                  #Temperature
    initialConcentrations={
        "decane": (4.70000000E-03,'mol/cm^3'),                   #fuel conc
        "oxygen": (2.00000000E-05,'mol/cm^3'),                   #O2 conc
        "ROOH": (1.00000000E-10,'mol/cm^3'),                          #ROOH conc
        "RO": (1.00000000E-10,'mol/cm^3'),                                    #RO conc
        "OH": (1.00000000E-10,'mol/cm^3')                                     #OH conc
    },
    terminationTime=(1200,'s'),            #time
)

solvation(
    solvent='decane'                              #solvent
)

simulator(
    atol=1e-16,
    rtol=1e-8,
)

model(
    toleranceMoveToCore=0.1,
    toleranceInterruptSimulation=0.1,
)

options(
    units='si',
    saveRestartPeriod=False,
    generateOutputHTML=True,
    generatePlots=True,
    saveSimulationProfiles=True,
)

**Comment by mjohnson541** [comment]
You don't have to include all of the species in your input file in the reactor block, if you leave them out they'll just be zero at the start of the simulation. I think your problem is toleranceMoveToCore, liquidReactor often requires much lower tolerances than gas phase reactors. Unless the runs are taking prohibitively long (usually if toleranceMoveToCore is too large the run ends pretty fast) I'd suggest reducing toleranceMoveToCore by an order of magnitude until it picks up the ROOH => RO + OH reaction.

**Comment by ysadat7** [comment]
Thank you Matt, I'll have a few runs with reducing TMTC. OH I SEE! So I don't have to assign them a concentration at all?

**Comment by mjohnson541** [comment]
Yep, if you include them they will start included in the model (although not necessarily reacted if filterReactions=True, but usually that isn't a problem). I would actually highly recommend setting `filterReactions=True` in the model block for any production runs, it speeds up RMG a lot for larger models and we've never seen it have undesirable behavior.

**Comment by ysadat7** [comment]
That’s very helpful! I’ll let you know of the outcome.

**Comment by ysadat7** [comment]
Hi again Matt

So I tried 0.01, 0.005 and 0.005 as TMTC values, there was some improvement in the prediction of Hydroperoxide profile (see attached compared to another paper with manual construction of the mechanism), https://drive.google.com/file/d/1TvhOk3LKr0NzAEoRUrpC_nYmOqv57hD9/view?usp=sharing

Before that it was just a straight line, however I still don't see the hydroperoxide decomposition in the list of reactions, particularly this one: ROOH --> RO' + OH' which is a key step reaction in the fuel autoxidation chain of reactions. Do you think it would be possible to manually add this reaction step in the seed mechanism and have a run like that?

I also tried **_filterreactions=true_** and it was significantly helpful with runs going faster. Leaving the species out of the reactor part was also very helpful.

Many thanks!

**Comment by mjohnson541** [comment]
Simplest way to do that should be to include the species in the input file, if that isn't sufficient you can include a react block that explicitly reacts all of the tuples of species names together:

```
react([
("H","OH"),
("CH4",),
("ROOH",)])
```

This block will react the species with name "H" with the species named "OH", react the species name "CH4"  by itself, and react the species named "ROOH" by itself.

**Comment by mjohnson541** [comment]
You can definitely add it as a part of a seed mechanism, that just requires you to write the seed mechanism and assign the rate, if you go that direction you can use our web tool for estimation here: https://rmg.mit.edu/

**Comment by ysadat7** [comment]
I have been trying to incorporate the react( in my input file but the runs keep failing, would you please take a look at my input file please? Many thanks

database(
    thermoLibraries = ['primaryThermoLibrary'],
    reactionLibraries = [],
    seedMechanisms = [],
    kineticsDepositories = ['training'],
    kineticsFamilies = 'default',
    kineticsEstimator = 'rate rules',
)

# List of species                            fuel, ao, and smiles

species(
    label='decane',
    reactive=True,
    structure=SMILES("CCCCCCCCCC"),
)
species(
    label='oxygen',
    reactive=True,
    structure=SMILES("[O][O]"),
)
species(
    label='ROOH',
    reactive=True,
    structure=SMILES("CCCCCCCCCCOO"),
)
species(
    label='RO',
    reactive=True,
    structure=SMILES("CCCCCCCCCC[O]"),
)
species(
    label='OH',
    reactive=True,
    structure=SMILES("[OH]"),
)
species(
    label='ROH',
    reactive=True,
    structure=SMILES("CCCCCCCCCCO"),
)
species(
    label='R(=O)R',
    reactive=True,
    structure=SMILES("CCCCCCCC(=O)CC"),
)
species(
    label='R=O',
    reactive=True,
    structure=SMILES("CCCCCCCCCC=O"),
)

# Reaction systems
liquidReactor(
    temperature=(423,'K'),                  #Temperature
    initialConcentrations={
        "decane": (4.70000000E-03,'mol/cm^3'),                   #fuel conc
        "oxygen": (2.00000000E-06,'mol/cm^3'),                   #O2 conc
        "ROOH": (2.00000000E-10,'mol/cm^3')                      #ROOH conc  
    },
    terminationTime=(1200,'s'),            #time
    
    **react(
    [("H","OH"),
    ("CH4",),
    ("ROOH",)]),**
)

solvation(
    solvent='decane'                              #solvent
)

simulator(
    atol=1e-16,
    rtol=1e-8,
)

model(
    toleranceMoveToCore=0.001,
    toleranceInterruptSimulation=0.1,
    filterReactions=True,
)

options(
    units='si',
    saveRestartPeriod=False,
    generateOutputHTML=True,
    generatePlots=True,
    saveSimulationProfiles=True,
)

**Comment by mjohnson541** [comment]
To use that you currently have to have installed RMG from source, that feature isn't available in the current binaries.

**Comment by ysadat7** [comment]
I see, is there a guide on how I could do that? I searched the
documentations and didn’t find anything. I’m not running RMG locally, would
that be a problem?

**Comment by mjohnson541** [comment]
That shouldn't be a problem. The directions are here: http://reactionmechanismgenerator.github.io/RMG-Py/users/rmg/installation/index.html#for-developers-installation-by-source-using-anaconda-environment
